packages/skills/weather_station_ledger/handlers.py

```diff
@@ -0,0 +1,262 @@
+
+"""This package contains a scaffold of a handler."""
+
+import logging
+import json
+import time
+from typing import Any, Dict, List, Optional, Union, cast, TYPE_CHECKING
+
+from fetchai.ledger.api import LedgerApi  # type: ignore
+from fetchai.ledger.crypto import Address, Identity  # type: ignore
+
+from aea.configurations.base import ProtocolId
+
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.protocols.fipa.message import FIPAMessage
+from aea.protocols.fipa.serialization import FIPASerializer
+from aea.protocols.oef.models import Description
+from aea.skills.base import Handler
+
+if TYPE_CHECKING:
+    from packages.skills.weather_station.db_communication import DBCommunication
+else:
+    from weather_station_skill.db_communication import DBCommunication
+
+logger = logging.getLogger("aea.weather_station_skill")
+
+DATE_ONE = "3/10/2019"
+DATE_TWO = "15/10/2019"
+
+
+class MyWeatherHandler(Handler):
+    """This class scaffolds a handler."""
+
+    SUPPORTED_PROTOCOL = FIPAMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initialise the behaviour."""
+        super().__init__(**kwargs)
+        self.fet_price = 0.002
+        self.db = DBCommunication()
+        self.fetched_data = []
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message, sender: str) -> None:
+        """
+        Implement the reaction to an message.
+
+        :param message: the message
+        :param sender: the sender
+        :return: None
+        """
+        fipa_msg = cast(FIPAMessage, message)
+        msg_performative = FIPAMessage.Performative(fipa_msg.get('performative'))
+        message_id = cast(int, fipa_msg.get('id'))
+        dialogue_id = cast(int, fipa_msg.get('dialogue_id'))
+
+        if msg_performative == FIPAMessage.Performative.CFP:
+            self.handle_cfp(fipa_msg, sender, message_id, dialogue_id)
+        elif msg_performative == FIPAMessage.Performative.ACCEPT:
+            self.handle_accept(sender)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def handle_cfp(self, msg: FIPAMessage, sender: str, message_id: int, dialogue_id: int) -> None:
+        """
+        Handle the CFP calls.
+
+        :param msg: the message
+        :param sender: the sender
+        :param message_id: the message id
+        :param dialogue_id: the dialogue id
+        :return: None
+        """
+        new_message_id = 1
+        new_target = message_id
+        fetched_data = self.db.get_data_for_specific_dates(DATE_ONE, DATE_TWO)
+
+        if len(fetched_data) >= 1:
+            self.fetched_data = fetched_data
+            total_price = self.fet_price * len(fetched_data)
```

Comment:
the fet_price should also be expressed via preferences

Reply:
It can be expressed via preferences though, at the moment the decision-maker expresses the client and not the weather_station.